lms/templates/courseware/course_about.html

```diff
@@ -109,7 +109,11 @@
           location.href = xhr.responseText;
         }
       } else if (xhr.status == 403) {
-          location.href = "${reverse('register_user')}?course_id=${course.id | u}&enrollment_action=enroll";
+          %if settings.SHIB_ONLY_SITE:
+            location.href = "/login?next=${reverse('finish_auth') + '?course_id=' + course.id.to_deprecated_string() + '&enrollment_action=enroll&next=' + reverse('dashboard') | u}";
```

Comment:
Do we need the `course.id.to_deprecated_string`?

Reply:
Yes otherwise it errors because it can't convert a CourseLocator type object to unicode

Reply:
huh, but then why does the original not use `course.id.to_deprecated_string`?

Reply:
When there is no string concatenation it gets cast normally, the issue only arises with concatenation

Reply:
Gotcha